sysbuild: Add AP-Protect settings and documentation (#17836)
nrfconnect/sdk-nrf: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst

* Added the ``SB_CONFIG_MCUBOOT_NRF53_MULTI_IMAGE_UPDATE`` sysbuild Kconfig option that enables updating the network core on the nRF5340 SoC from external flash.

> * Added the AP-Protect sysbuild Kconfig options to enable corresponding AP-Protect Kconfigs for all the images in build:
>     * ``SB_CONFIG_APPROTECT_LOCK`` for :kconfig:option:`CONFIG_NRF_APPROTECT_LOCK` Kconfig option.
>     * ``SB_CONFIG_APPROTECT_USER_HANDLING`` for :kconfig:option:`CONFIG_NRF_APPROTECT_USER_HANDLING` Kconfig option.
>     * ``SB_CONFIG_APPROTECT_USE_UICR`` for :kconfig:option:`CONFIG_NRF_APPROTECT_USE_UICR` Kconfig option.
>     * ``SB_CONFIG_SECURE_APPROTECT_LOCK`` for :kconfig:option:`CONFIG_NRF_SECURE_APPROTECT_LOCK` Kconfig option.
>     * ``SB_CONFIG_SECURE_APPROTECT_USER_HANDLING`` for :kconfig:option:`CONFIG_NRF_SECURE_APPROTECT_USER_HANDLING` Kconfig option.
>     * ``SB_CONFIG_SECURE_APPROTECT_USE_UICR`` for :kconfig:option:`CONFIG_NRF_SECURE_APPROTECT_USE_UICR` Kconfig option.

Reviewer: ```suggestion
* Added AP-Protect sysbuild Kconfig options to enable the corresponding AP-Protect Kconfig options for all images in the build:

  * ``SB_CONFIG_APPROTECT_LOCK`` for :kconfig:option:`CONFIG_NRF_APPROTECT_LOCK` Kconfig option.
  * ``SB_CONFIG_APPROTECT_USER_HANDLING`` for :kconfig:option:`CONFIG_NRF_APPROTECT_USER_HANDLING` Kconfig option.
  * ``SB_CONFIG_APPROTECT_USE_UICR`` for :kconfig:option:`CONFIG_NRF_APPROTECT_USE_UICR` Kconfig option.
  * ``SB_CONFIG_SECURE_APPROTECT_LOCK`` for :kconfig:option:`CONFIG_NRF_SECURE_APPROTECT_LOCK` Kconfig option.
  * ``SB_CONFIG_SECURE_APPROTECT_USER_HANDLING`` for :kconfig:option:`CONFIG_NRF_SECURE_APPROTECT_USER_HANDLING` Kconfig option.
  * ``SB_CONFIG_SECURE_APPROTECT_USE_UICR`` for :kconfig:option:`CONFIG_NRF_SECURE_APPROTECT_USE_UICR` Kconfig option.
```

* Added the ``SB_CONFIG_MCUBOOT_NRF53_MULTI_IMAGE_UPDATE`` sysbuild Kconfig option that enables updating the network core on the nRF5340 SoC from external flash.

* Added AP-Protect sysbuild Kconfig options to enable the corresponding AP-Protect Kconfig options for all images in the build:

>   * ``SB_CONFIG_APPROTECT_LOCK`` for :kconfig:option:`CONFIG_NRF_APPROTECT_LOCK` Kconfig option.

Reviewer 2: ```suggestion
  * ``SB_CONFIG_APPROTECT_LOCK`` for the :kconfig:option:`CONFIG_NRF_APPROTECT_LOCK` Kconfig option.
```

  * The :ref:`dfu_tools_mcumgr_cli` page after it was removed from the Zephyr repository.
  * The :ref:`ug_nrf54h20_suit_soc_binaries` page.
  * The :ref:`ug_nrf54h20_suit_push` page documentating the SUIT push model-based update process.

>   * The :ref:`app_secure_approtect` section to the :ref:`app_approtect` page.

Reviewer 2: There is more than just one new section.

Author: The other sections are subsection of this.

  * The :ref:`programming_params` section on the :ref:`programming` page with information about readback protection moved from the :ref:`ug_nrf5340_building` page.
  * The :ref:`security` page with a table that provides an overview of the available general security features.
    This table replaces the subpage that was previously describing these features in more detail and was duplicating information available in other sections.

>   * The :ref:`app_approtect_ncs` section on the :ref:`app_approtect` page with details on setting the Kconfigs and register values to enable AP-Protect.

Reviewer: ```suggestion
  * The :ref:`app_approtect_ncs` section on the :ref:`app_approtect` page with details on setting the Kconfig options and register values to enable AP-Protect.
```
